internal/pkg/experimental/action/catalog_inspect.go

```
@@ -0,0 +1,160 @@
+	// contents should be from.
+	// Optional.
+	Package string
+	// Output is the format that contents should be
+	// written to STDOUT in.
+	// Optional.
+	Output string
+
+	// Name is the name of the catalog object to inspect. Required.
+	Name string
+}
+
+const OutputFlag = "output"
+
+// BindCatalogInspectFlags binds the flags for the "catalog inspect"
+// subcommand to the provided pflag.FlagSet
+func BindCatalogInspectFlags(flagSet *pflag.FlagSet) {
+	flagSet.String(CatalogFlag, "", "filter results to only be from the specified catalog")
+	flagSet.String(SchemaFlag, "", "filter results to only be FBC objects that have the specified schema")
+	flagSet.String(PackageFlag, "", "filter results to only be FBC objects that belong to the specified package")
+	flagSet.String(OutputFlag, "json", "the format in which output should be. One of [json, yaml]")
+}
+
+// CatalogInspectOptionsFromFlags will return a CatalogInspectOptions pointer with the options
+// fetched from the values for the corresponding flags in a given pflag.FlagSet. Returns nil and
+// an error if any errors are encountered during the process
+func CatalogInspectOptionsFromFlags(flagSet *pflag.FlagSet) (*CatalogInspectOptions, error) {
+	catalog, err := flagSet.GetString(CatalogFlag)
+	if err != nil {
+		return nil, fmt.Errorf("parsing catalog flag: %w", err)
+	}
+
+	schema, err := flagSet.GetString(SchemaFlag)
+	if err != nil {
+		return nil, fmt.Errorf("parsing schema flag: %w", err)
+	}
+
+	pack, err := flagSet.GetString(PackageFlag)
+	if err != nil {
+		return nil, fmt.Errorf("parsing package flag: %w", err)
+	}
+
+	output, err := flagSet.GetString(OutputFlag)
+	if err != nil {
+		return nil, fmt.Errorf("parsing output flag: %w", err)
+	}
+
+	return &CatalogInspectOptions{
+		Catalog: catalog,
+		Schema:  schema,
+		Package: pack,
+		Output:  output,
+	}, nil
+}
+
+func NewCatalogInspect(cfg *action.Configuration, fetcherFunc FetcherFunc, streamerFunc StreamerFunc) (*CatalogInspect, error) {
```

Don't need to return an error in this function definition.

---

Ah good catch, this was from when I was doing some setup that could error and  i forgot to clean it up 🤦